vital/types/geodesy.cxx

```diff
@@ -0,0 +1,67 @@
+/*ckwg +29
+ * Copyright 2017 by Kitware, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ *  * Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  * Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ *  * Neither name of Kitware, Inc. nor the names of any contributors may be used
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief This file contains the implementation of a geo point.
+ */
+
+#include "geodesy.h"
+
+#include <atomic>
+
+namespace kwiver {
+namespace vital {
+
+namespace {
+  static std::atomic< geo_conversion* > s_geo_conv;
+}
+
+// ----------------------------------------------------------------------------
+void
+set_geo_conv( geo_conversion* c )
+{
+  s_geo_conv.store( c );
```

Comment:
> So if I have multiple plugins that provide geo-conversion then which ever is loaded last wins, right?

"Yes". However...

>  It would be much nicer if [...] there was a way to store multiple loaded options and configure which plugin is active if multiple are provided.

...you can re-register the conversion functor yourself.

I agree there is room for improvement here, but I'm not sure it's worth spending the time now when there is only one implementation available. Also, this isn't like other algorithms where different implementations will give totally different results. Two correct implementations should be, besides timing, nearly indistinguishable to the user.

Related to the first point, where I had a similar situation in ViViA, I solved it by having each plugin be assigned a priority, such that whichever has a higher priority "wins".